pkg/devices.go

```diff
@@ -0,0 +1,210 @@
+package pkg
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	cdispec "github.com/container-orchestrated-devices/container-device-interface/specs-go"
+	spec "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	root = "/etc/cdi"
+)
+
+var (
+	defaultLogger = logrus.New()
```

Comment:
We could avoid logging if we want to use this as a library. The callers could be returned errors and decide what to do.